### ISSUE TITLE
how to finetune with gemma model？

### ISSUE THREAD

**Issue by runningabcd** [google/gemma_pytorch#8]
how to finetune with gemma model？

**Comment by runningabcd** [comment]
help

**Comment by runningabcd** [comment]
> I had download gemma-7b-it model from hugging face already, but not find the script that can finetune wiht my own data

How to sft with gemma？Can you tell me the sft data format?

**Comment by runningabcd** [comment]
> > I had download gemma-7b-it model from hugging face already, but not find the script that can finetune wiht my own data
> 
> How to sft with gemma？Can you tell me the sft data format?

@pengchongjin

**Comment by yongzhuo** [comment]
e......

I have tried [https://github.com/yongzhuo/gemma-sft](https://github.com/yongzhuo/gemma-sft), but sometimes NAN, maybe not stable.

logs:
{'loss': 0.0401, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0378, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0339, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0411, 'lr': 0.0003, 'epoch': 0.76}
{'loss': nan, 'lr': 0.0003, 'epoch': 0.76}
{'loss': nan, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0436, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0336, 'lr': 0.0003, 'epoch': 0.76}
{'loss': nan, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0514, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.05, 'lr': 0.0003, 'epoch': 0.76}
{'loss': 0.0696, 'lr': 0.0003, 'epoch': 0.76}
{'loss': nan, 'lr': 0.0003, 'epoch': 0.76}

**Comment by r-gheda** [comment]
@pengchongjin Is it possible to implement a class for fine tuning the model similar to what done with llama-recipes?

**Comment by aliasneo1** [comment]
Are there any tutorials for fine-tuning 7b-it-quant model ?

**Comment by selamw1** [comment]
Hi @aliasneo1 

There are a few tutorials that demonstrate fine-tuning the `gemma-2b` model. You can follow similar procedures to fine-tune the Gemma variant `gemma-7b-it`.

Here are some resources:

- [Fine-tuning Gemma](https://ai.google.dev/gemma/docs/jax_finetune) using JAX and Flax.
- [Fine-Tuning Gemma](https://huggingface.co/blog/gemma-peft) Models in Hugging Face with PyTorch on GPU and TPU.

**Comment by Gopi-Uppari** [comment]
Hi @runningabcd,

Could you please confirm if this issue is resolved for you with the above comment ? Please feel free to close the issue if it is resolved ?

Thank you.

**Comment by Gopi-Uppari** [comment]
Hi @runningabcd,

Closing this issue due to lack of recent activity,  Please reopen if this is still a valid request. 

Thank you!
